terraform-provider/internal/service/subgraph/data_source_cosmo_subgraph.go

```diff
@@ -0,0 +1,169 @@
+package subgraph
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/wundergraph/cosmo/terraform-provider-cosmo/internal/api"
+	"github.com/wundergraph/cosmo/terraform-provider-cosmo/internal/client"
+	"github.com/wundergraph/cosmo/terraform-provider-cosmo/internal/utils"
+)
+
+var _ datasource.DataSource = &SubgraphDataSource{}
+
+func NewSubgraphDataSource() datasource.DataSource {
+	return &SubgraphDataSource{}
+}
+
+type SubgraphDataSource struct {
+	*client.PlatformClient
+}
+
+type SubgraphDataSourceModel struct {
+	Id                   types.String `tfsdk:"id"`
+	Name                 types.String `tfsdk:"name"`
+	Namespace            types.String `tfsdk:"namespace"`
+	RoutingUrl           types.String `tfsdk:"routing_url"`
+	BaseSubgraphName     types.String `tfsdk:"base_subgraph_name"`
+	Labels               types.List   `tfsdk:"labels"`
+	SubscriptionUrl      types.String `tfsdk:"subscription_url"`
+	SubscriptionProtocol types.String `tfsdk:"subscription_protocol"`
+	Readme               types.String `tfsdk:"readme"`
+	WebsocketSubprotocol types.String `tfsdk:"websocket_subprotocol"`
+	IsEventDrivenGraph   types.Bool   `tfsdk:"is_event_driven_graph"`
+	IsFeatureSubgraph    types.Bool   `tfsdk:"is_feature_subgraph"`
+	Headers              types.List   `tfsdk:"headers"`
+}
+
+func (d *SubgraphDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_subgraph"
+}
+
+func (d *SubgraphDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Cosmo Subgraph Data Source",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The unique identifier of the subgraph resource.",
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The name of the subgraph.",
+			},
+			"namespace": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The namespace in which the subgraph is located.",
+			},
+			"routing_url": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The routing URL of the subgraph.",
+			},
+			"base_subgraph_name": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The base subgraph name.",
```

Comment:
This is not corrrect.

Reply:
Fixed in the new repository.

The value is not exposed in the schema anymore and can't be accessed.